src/main/java/org/jabref/gui/externalfiles/ExternalFilesEntryLinker.java

```diff
@@ -102,4 +110,32 @@ public void copyFilesToFileDirAndAddToEntry(BibEntry entry, List<Path> files, In
             LOGGER.error("Could not access Fulltext-Index", e);
         }
     }
+
+    private List<Path> getValidFileNames(List<Path> filesToAdd) {
+        List<Path> validFileNames = new ArrayList<>();
+
+        for (Path fileToAdd : filesToAdd) {
+            if (FileUtil.detectBadFileName(fileToAdd.toString())) {
+                String newFilename = FileNameCleaner.cleanFileName(fileToAdd.getFileName().toString());
+
+                boolean correctButtonPressed = dialogService.showConfirmationDialogAndWait(Localization.lang("File \"%0\" cannot be added!", fileToAdd.getFileName()),
+                        Localization.lang("Illegal characters in the file name detected.\nFile will be renamed to \"%0\" and added.", newFilename),
+                        Localization.lang("Rename and add"));
+
+                if (correctButtonPressed) {
+                    Path correctPath = fileToAdd.resolveSibling(newFilename);
+                    try {
+                        Files.move(fileToAdd, correctPath);
+                        validFileNames.add(correctPath);
+                    } catch (IOException ex) {
+                        LOGGER.error("Error moving file", ex);
+                        dialogService.showErrorDialogAndWait(ex);
+                    }
+                }
+            } else {
+                validFileNames.add(fileToAdd);
+            }
+        }
+        return validFileNames;
+    }
```

Comment:
A lot of code duplication to LinkedFilesEditorViewModel, but fine for a hotfix. Maybe an followup PR should clean this up.